e2e/exec.go

```diff
@@ -100,10 +100,22 @@ func sshString(vmPrivateIP string) string {
 	return fmt.Sprintf(`ssh -i %s -o PasswordAuthentication=no -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ConnectTimeout=5 azureuser@%s`, sshKeyName(vmPrivateIP), vmPrivateIP)
 }
 
+func quoteForBash(command string) string {
+	return fmt.Sprintf("'%s'", strings.ReplaceAll(command, "'", "'\"'\"'"))
```

Comment:
it feels like voodoo magic. I wonder if there is a function from standard SDK to achieve similar results

Reply:
Is it possible to use something like this? 
```
ssh user@remote "bash -s" <<EOF
echo "Hello, world!"
echo 'This script contains "quotes" and other characters.'
EOF
```

Reply:
I'm actually working on something like this. Was going to make it a separate PR, but will add it to this PR.